server/base/src/main/java/org/apache/accumulo/server/security/handler/ZKPermHandler.java

```diff
@@ -396,11 +398,14 @@ public void initializeSecurity(TCredentials itw, String rootuser)
     tablePerms.put(RootTable.ID, Collections.singleton(TablePermission.ALTER_TABLE));
     tablePerms.put(MetadataTable.ID, Collections.singleton(TablePermission.ALTER_TABLE));
     // essentially the same but on the system namespace, the ALTER_TABLE permission is now redundant
+    // After PR #2994 which added security checks for configuration we need to also add
+    // ALTER_NAMESPACE
+    // to both Default and Accumulo Namespaces for the root user.
     Map<NamespaceId,Set<NamespacePermission>> namespacePerms = new HashMap<>();
-    namespacePerms.put(Namespace.ACCUMULO.id(),
+    namespacePerms.put(Namespace.DEFAULT.id(),
```

Comment:
I think this needs to be removed and the comment udpated.

Reply:
You can't remove this or it doesn't work. You have to grant ALTER_NAMESPACE to the default namespace or else when you use the config command as the root user on a table in the default namespace you get a permission error.

Reply:
I should add you don't need to grant READ permission or anything else but you specifically need ALTER_NAMESPACE

Reply:
Yes, I know. But you are handling the permission errors in part 2.

Reply:
Right, but I figured we'd still want to give permission to see the configs on tables in the default namespace for the root user (but maybe not). I know @ctubbsii [said](https://github.com/apache/accumulo/pull/2994#discussion_r986243402) to drop READ permission but I thought permission for the configs made sense since the root user is probably an admin trying to configure things so would want to see configs even if can't scan the table by default.

Reply:
That's fair, hopefully @ctubbsii can chime in on this.

Reply:
> You can't remove this or it doesn't work. You have to grant ALTER_NAMESPACE to the default namespace or else when you use the config command as the root user on a table in the default namespace you get a permission error.

Yeah, that's the desired behavior. The root user does not get permissions by default to any user namespaces, including the default namespace (which is really only a legacy namespace intended for user tables, to preserve the ability to specify an unqualified table name, like before the namespace feature was added).